### ISSUE TITLE
Replace archived atom node-keytar

### ISSUE THREAD

**Issue by Johennes** [vector-im/element-web#24444]
https://freek.dev/2103-replacing-keytar-with-electrons-safestorage-in-ray
https://www.electronjs.org/docs/latest/api/safe-storage

**Comment by Johennes** [comment]
> The only downside is that any previously saved passwords and passphrases saved using Keytar are now inaccessible for the app

Is there a way to migrate previously stored secrets?

**Comment by t3chguy** [comment]
@Johennes yes, but it'll mean keeping the keytar dependency indefinitely otherwise users might miss the migration.

**Comment by Johennes** [comment]
@t3chguy what impact would it have on the user if we didn't migrate their previous secrets, meaning what do we currently store via keytar?

**Comment by t3chguy** [comment]
> @t3chguy what impact would it have on the user if we didn't migrate their previous secrets, meaning what do we currently store via keytar?

Users would lose their sessions, we pickle a lot of things in IDB based on what's stored via keytar.

**Comment by Johennes** [comment]
Summary from refinement:

We can't force users to go through a specific version that would guarantee their secrets to be migrated because Squirrel doesn't support that feature.

We also can't properly warn users about losing their secrets before it's too late either, again because Squirrel doesn't offer this feature.

We also don't support downgrades, so once it is too late, there is no way to go back.

Finally, we can't migrate to something else than Squirrel because Electron's autoupdater only supports squirrel on Mac.

Moreover, there are things like chocolatey and brew packages where we'd have virtually no control over the update process.

Writing manual migration code that doesn't depend on keytar seems like a minefield given the various keyring differences between platforms.

In total that only seems to leave us with two options:

* Adopt maintenance over node-keytar
* Define a reasonable cut-off and consciously accept people losing their secrets if they didn't upgrade in X months

**Comment by Johennes** [comment]
Summary from today's refinement:

- It might be possible to create an external app to read the old keys but that'll not work on all systems
- We could leverage dehydrated devices but not all users allow 4S

The most reasonable step forward currently appears to be to implement the migration from node-keytar to `safeStorage` but to _not_ remove node-keytar until we hit an actual problem with it. This way new and updated installations stop depending on node-keytar but we retain the backwards support as long as possible without locking anyone out prematurely.
